### PR TITLE
[SPARK-35708][PYTHON][TEST] Add BaseTest for DataTypeOps

### PR DESCRIPTION
**Pull request by Yikun**
### What changes were proposed in this pull request?
This patch adds DataTypeOps test to check the ops is loaded as expected. 

### Why are the changes needed?
When complete https://github.com/apache/spark/pull/32821, I found there are no test for DataTypeOps. There were many logic when DataTypeOps loaded, it's better to add the test to make sure interface stable. 

### Does this PR introduce _any_ user-facing change?
No, test only

### How was this patch tested?
test passed.

### CODE REVIEW

**Comment by Yikun** [comment]
cc @xinrong-databricks @HyukjinKwon

**Comment by xinrong-databricks** [comment]
Looks great! Thanks.

**Review by xinrong-meng — APPROVED** [review]

**Comment by xinrong-databricks** [comment]
CC @ueshin @itholic

**Comment by xinrong-databricks** [comment]
Reminder: We might include `DecimalOps` in the test according to which PR (this or #32821) is merged first.

File: python/pyspark/pandas/tests/data_type_ops/test_base.py
Comment (ueshin) [inline python/pyspark/pandas/tests/data_type_ops/test_base.py:20]: This will raise an `ImportError` if the nullable boolean type is not available (`pandas<1.0`).
Could you also test with the old pandas?

File: python/pyspark/pandas/tests/data_type_ops/test_base.py
Comment (ueshin) [inline python/pyspark/pandas/tests/data_type_ops/test_base.py:21]: Shall we split the import block between external libraries (`pandas` here) and our library?

File: python/pyspark/pandas/tests/data_type_ops/test_base.py
Comment (Yikun) [reply python/pyspark/pandas/tests/data_type_ops/test_base.py:20]: done

File: python/pyspark/pandas/tests/data_type_ops/test_base.py
Comment (Yikun) [reply python/pyspark/pandas/tests/data_type_ops/test_base.py:21]: Sure

**Comment by Yikun** [comment]
> Reminder: We might include DecimalOps in the test according to which PR (this or #32821) is merged first.

@xinrong-databricks Yep. I'm going to update DecimalOps test case after this merge.

File: python/pyspark/pandas/tests/data_type_ops/test_base.py
Comment (ueshin) [inline python/pyspark/pandas/tests/data_type_ops/test_base.py:27]: This is imported twice. I think we can put `from pandas import BooleanDtype` in `test_bool_ext_ops`.

File: python/pyspark/pandas/tests/data_type_ops/test_base.py
Comment (ueshin) [inline python/pyspark/pandas/tests/data_type_ops/test_base.py:58]: I guess we don't need to extend both `PandasOnSparkTestCase` and `TestCasesUtils` but only `unittest.TestCase` for this test case?

File: python/pyspark/pandas/tests/data_type_ops/test_base.py
Comment (ueshin) [inline python/pyspark/pandas/tests/data_type_ops/test_base.py:87]: The test cases in this file is not triggered. Could you add a footer?

Comment (Yikun) [reply python/pyspark/pandas/tests/data_type_ops/test_base.py:87]: Oops, thanks

File: python/pyspark/pandas/tests/data_type_ops/test_base.py
Comment (Yikun) [reply python/pyspark/pandas/tests/data_type_ops/test_base.py:58]: yep

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
@Yikun can you retrigger https://github.com/Yikun/spark/actions/runs/948966796?

**Comment by ueshin** [comment]
@Yikun now that #32821 is merged, could you update to add `DecimalOps` test case?

**Comment by Yikun** [comment]
> @Yikun now that #32821 is merged, could you update to add DecimalOps test case?

@ueshin Thanks for reminder, done. : )

**Review by ueshin — APPROVED** [review]
LGTM, pending tests.

**Comment by ueshin** [comment]
Thanks! merging to master.

**Pull request closed without merging**
